pkg/metrics/metrics.go

```diff
@@ -1,143 +1,328 @@
-		},
-		policyChangesLabels,
+
+	res, err := resource.New(context.Background(),
+		resource.WithAttributes(semconv.ServiceNameKey.String("kyverno_metrics")),
+		resource.WithSchemaURL(semconv.SchemaURL),
 	)
+	if err != nil {
+		log.Error(err, "failed creating resource")
+	}
 
-	policyExecutionDurationLabels := []string{
-		"policy_validation_mode", "policy_type", "policy_background_mode", "policy_name", "policy_namespace",
-		"resource_kind", "resource_namespace", "resource_request_operation",
-		"rule_name", "rule_result", "rule_type", "rule_execution_cause", "generate_rule_latency_type",
-	}
-	policyExecutionDurationMetric := prom.NewHistogramVec(
-		prom.HistogramOpts{
-			Name: "kyverno_policy_execution_duration_seconds",
-			Help: "can be used to track the latencies (in seconds) associated with the execution/processing of the individual rules under Kyverno policies whenever they evaluate incoming resource requests.",
-		},
-		policyExecutionDurationLabels,
+	// create controller and bind the exporter with it
+	pusher := controller.New(
+		processor.NewFactory(
+			simple.NewWithHistogramDistribution(),
+			aggregation.CumulativeTemporalitySelector(),
+			processor.WithMemory(true),
+		),
+		controller.WithExporter(metricExp),
+		controller.WithResource(res),
+		controller.WithCollectPeriod(2*time.Second),
 	)
+	global.SetMeterProvider(pusher)
+
+	m := new(MetricsConfig)
+	m.Log = log
+	m.Config = metricsConfigData
 
-	admissionReviewDurationLabels := []string{
-		"resource_kind", "resource_namespace", "resource_request_operation",
+	m, err = initializeMetrics(m)
+
+	if err != nil {
+		log.Error(err, "Failed initializing metrics")
+		return nil, err
 	}
-	admissionReviewDurationMetric := prom.NewHistogramVec(
-		prom.HistogramOpts{
-			Name: "kyverno_admission_review_duration_seconds",
-			Help: "can be used to track the latencies (in seconds) associated with the entire individual admission review. For example, if an incoming request trigger, say, five policies, this metric will track the e2e latency associated with the execution of all those policies.",
-		},
-		admissionReviewDurationLabels,
-	)
 
-	admissionRequestsLabels := []string{
-		"resource_kind", "resource_namespace", "resource_request_operation",
+	if err := pusher.Start(ctx); err != nil {
+		log.Error(err, "could not start metric exporter")
+		return nil, err
 	}
-	admissionRequestsMetric := prom.NewCounterVec(
-		prom.CounterOpts{
-			Name: "kyverno_admission_requests_total",
-			Help: "can be used to track the number of admission requests encountered by Kyverno in the cluster.",
-		},
-		admissionRequestsLabels,
+
+	return m, nil
+}
+
+func NewPrometheusConfig(metricsConfigData *config.MetricsConfigData,
+	log logr.Logger,
+) (*MetricsConfig, *http.ServeMux, error) {
+
+	config := prometheus.Config{}
+	res, err := resource.New(context.Background(),
+		resource.WithAttributes(semconv.ServiceNameKey.String("kyverno-svc-metrics")),
```

Comment:
Is the service name fixed? If so, needs to call ou in the doc.

Reply:
Service name is fixed for now. Will update in docs